cmds/modules/noded/main.go

```
@@ -297,12 +228,7 @@ func action(cli *cli.Context) error {
 
 	server.Register(zbus.ObjectID{Name: "host", Version: "0.0.1"}, host)
 	server.Register(zbus.ObjectID{Name: "system", Version: "0.0.1"}, system)
-
-	go func() {
-		if err := server.Run(ctx); err != nil && err != context.Canceled {
-			log.Fatal().Err(err).Msg("unexpected error")
-		}
-	}()
+	server.Register(zbus.ObjectID{Name: "host", Version: "0.0.1"}, perfMon)
```

isn't this a double register of the object `host` ?

---

check the return error from register. this should return an error because you trying to register an object with the same id